test/UnitTests/MSTest.CoreAdapter.Unit.Tests/Execution/TestExecutionManagerTests.cs

```diff
@@ -148,13 +148,9 @@ public void RunTestsForASingleTestShouldSendSingleResult()
 
             this.TestExecutionManager.RunTests(tests, this.runContext, this.frameworkHandle, new TestRunCancellationToken());
 
-            List<string> expectedTestCaseStartList = new List<string>() { "PassingTest" };
-            List<string> expectedTestCaseEndList = new List<string>() { "PassingTest:Passed" };
-            List<string> expectedResultList = new List<string>() { "PassingTest  Passed" };
-
-            CollectionAssert.AreEqual(expectedTestCaseStartList, this.frameworkHandle.TestCaseStartList);
-            CollectionAssert.AreEqual(expectedTestCaseEndList, this.frameworkHandle.TestCaseEndList);
-            CollectionAssert.AreEqual(expectedResultList, this.frameworkHandle.ResultsList);
+            CollectionAssert.Contains(this.frameworkHandle.TestCaseStartList, "PassingTest");
```

Comment:
@jayaranigarg I added back the start/end events in TestExecutionManager. The side effect is that we now have duplicate start/end events for non-data driven tests. It's a bit odd but shouldn't effect execution because vstest ignores duplicate events. I don't think there is much I can do here because TestMethodInfo doesn't really  know if the test is being executed more than once.